integrations/Matomo/util.js

```diff
@@ -0,0 +1,295 @@
+      let keyword;
+      let category;
+      let resultsCount;
+      let contentInteraction;
+      let contentName;
+      let contentPiece;
+      let contentTarget;
+      let domId;
+
+      switch (val) {
+        case "trackLink":
+          url = get(properties, "url");
+          linkType = get(properties, "linkType");
+          window._paq.push(["trackLink", url, linkType]);
+          break;
+
+        case "trackSiteSearch":
+          keyword = get(properties, "keyword");
+          category = get(properties, "category");
+          resultsCount = get(properties, "resultsCount");
+          window._paq.push([
+            "trackSiteSearch",
+            keyword,
+            category,
+            resultsCount,
+          ]);
+          break;
+
+        case "ping":
+          window._paq.push(["ping"]);
+          break;
+
+        case "trackContentImpressionsWithinNode":
+          domId = get(properties, "domId") || get(properties, "dom_id");
+          window._paq.push([
+            "trackContentImpressionsWithinNode",
+            document.getElementById(domId),
+          ]);
+          break;
+
+        case "trackContentInteractionNode":
+          domId = get(properties, "domId") || get(properties, "dom_id");
+          contentInteraction = get(properties, "contentInteraction");
+          window._paq.push([
+            "trackContentInteractionNode",
+            document.getElementById(domId),
+            contentInteraction,
+          ]);
+          break;
+
+        case "trackContentImpression":
+          contentName = get(properties, "contentName");
+          contentPiece = get(properties, "contentPiece");
+          contentTarget = get(properties, "contentTarget");
+          window._paq.push([
+            "trackContentImpression",
+            contentName,
+            contentPiece,
+            contentTarget,
+          ]);
+          break;
+
+        case "trackContentInteraction":
+          contentInteraction = get(properties, "contentInteraction");
+          contentName = get(properties, "contentName");
+          contentPiece = get(properties, "contentPiece");
+          contentTarget = get(properties, "contentTarget");
+          window._paq.push([
+            "trackContentInteraction",
+            contentInteraction,
+            contentName,
+            contentPiece,
+            contentTarget,
+          ]);
+          break;
+
+        default:
+          break;
+      }
+    }
+  }, standardTo);
+};
+
+/** Mapping Ecommerce Events
+ */
+const ecommerceEventsMapping = (event, message) => {
+  let productSKU;
+  let productName;
+  let categoryName;
+  let price;
+  let quantity;
+  let orderId;
+  let grandTotal;
+  let subTotal;
+  let tax;
+  let shipping;
+  let discount;
+  let category;
+  let action;
+  let name;
+  let value;
+  const products = get(message, "properties.products"); // used for TRACK_ECOMMERCE_ORDER event
+
+  switch (event) {
+    case "SET_ECOMMERCE_VIEW":
```

Comment:
This is a `Product Viewed` event mapping? We hsould map RS Ecom Spec to Matomo Ecomm spec

Reply:
We should also use scoped case 
```
case SOMETHING: {
    code ...
    code ..
}

Reply:
Not needed actually as we have function scoped vars